x/validator-preference/msg_server.go

```diff
@@ -0,0 +1,68 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/osmosis-labs/osmosis/v12/x/validator-preference/types"
+)
+
+type msgServer struct {
+	keeper *Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper *Keeper) types.MsgServer {
+	return &msgServer{
+		keeper: keeper,
+	}
+}
+
+var _ types.MsgServer = msgServer{}
+
+func (server msgServer) SetValidatorSetPreference(goCtx context.Context, msg *types.MsgSetValidatorSetPreference) (*types.MsgSetValidatorSetPreferenceResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	preferences := msg.Preferences
+
+	// check if a user already have a validator-set created
+	existingValidators, found := server.keeper.GetValidatorSetPreference(ctx, msg.Delegator)
+
+	if found {
+		// check if the new preferences is the same as the existing preferences
+		if reflect.DeepEqual(preferences, existingValidators.Preferences) {
+			return nil, fmt.Errorf("The preferences are the same")
+		}
+
+		preferences = existingValidators.Preferences
```

Comment:
I think this is incorrect right? Isn't this saying "if user has existing preferences that were not the same as before, set preferences to what it was before", right?

Reply:
oh yea thats right! 

I dont think this is needed here as all the existing val-set are already validated!